src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,158 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+/**
+ * Handles the parsing of user input commands and creates corresponding Command objects.
+ * This class is responsible for interpreting user input and generating appropriate
+ * Command objects based on the input.
+ */
+public class Parser {
+
+    /**
+     * Parses the given input and returns the corresponding Command object.
+     *
+     * @param input The user input to be parsed.
+     * @return A Command object representing the parsed user input.
+     * @throws DukeException If the input cannot be parsed or contains errors.
+     */
+    public static Command parse(String input) throws DukeException {
+        if (input.equals("bye")) {
+            return new ExitCommand();
+        } else if (input.equals("list")) {
+            return new ListCommand();
+        } else if (input.contains("find")) {
+            String subInput;
+            try {
+                subInput = input.substring(5);
+                if (subInput.trim().equals("")) {
+                    throw new DukeException("The keyword to find cannot be empty.");
+                }
+            } catch (Exception e) {
+                throw new DukeException("The keyword to find cannot be empty.");
+            }
+            return new FindCommand(subInput);
+        } else if (input.contains("todo")) {
```

Comment:
Perhaps you can do input.startsWith("todo")? Same for the others too. Both works though!